tests/MongoDB.Driver.Core.Tests/Core/Operations/EndTransactionOperationTests.cs

```diff
@@ -93,17 +96,33 @@ public void CommandName_should_return_expected_result()
             result.Should().BeSameAs("fakeOperation");
         }
 
+        [Fact]
+        public void RecoveryToken_should_return_expected_result()
+        {
+            var recoveryToken = BsonDocument.Parse("{hp: 42}");
+            var subject = CreateSubject(recoveryToken: recoveryToken);
+
+            var result = subject.RecoveryToken;
+
+            result.Should().BeSameAs(recoveryToken);
+        }
+
         [Theory]
         [ParameterAttributeData]
         public void CreateCommand_should_return_expected_result(
+            [Values(true, false)] bool useRecoveryToken,
             [Values(1, 2)] int w)
         {
             var writeConcern = new WriteConcern(w);
-            var subject = CreateSubject(writeConcern: writeConcern);
+            var recoveryToken = BsonDocument.Parse("{cake: false}");
+            var subject = useRecoveryToken ? CreateSubject(recoveryToken, writeConcern) : CreateSubject(writeConcern: writeConcern);
 
             var result = subject.CreateCommand();
 
-            result.Should().Be($"{{ fakeOperation : 1, writeConcern : {{ w : {w} }} }}");
+            var expected = useRecoveryToken
+                ? $"{{ fakeOperation : 1, writeConcern : {{ w : {w} }}, recoveryToken : {recoveryToken} }}"
```

Comment:
Just small markup style suggestion:
I'm trying to set a space between brackets in `json`s for example:
https://github.com/DmitryLukyanov/mongo-csharp-driver/blob/a0a96eb3e5203b8dcd45c1b60d5ab0a6adee1a06/tests/MongoDB.Driver.Examples/DocumentationExamples.cs#L1477

So, I guess it looks a little bit better. Again it's up to you.

Reply:
Updated the recovderyToken spacing :)

Reply:
In the C# driver code we add a space before the colon in JSON also:

    { cake : false }